### DISCUSSION TITLE
Dynamic name attributes

### DISCUSSION THREAD

**Discussion by ragnaroktomorrow** [WerWolv/ImHex#818] [Pattern Language Q&A]
Is there a way to use the name attribute dynamically? Using constant, hard coded strings works ([[name("name")]], but whenever I use anything else in the parentheses, such as a std::format() or a reference to string data, the interpreter responds with "Expected ']]' after attribute, got Separator (()". 

The reported syntax error isn't quite accurate since even when using a constant string, the name attribute will be followed by an open-parentheses. This makes me wonder if it might already be possible to do this but the interpreter is just hitting an edge case since not all attributes have arguments.

Thanks :)

**Comment by vulpes2** [comment]
Related: https://github.com/WerWolv/PatternLanguage/issues/16
